src/dashboard/apps/consent/tests/test_models.py

```diff
@@ -1,63 +1,117 @@
+from django.db.models import signals
+from django.utils.timezone import make_aware
 
-from apps.auth.factories import UserFactory
 from apps.consent import AWAITING, REVOKED, VALIDATED
-from apps.consent.factories import ConsentFactory
+from apps.consent.signals import handle_new_delivery_point
+from apps.consent.utils import consent_end_date
 from apps.core.factories import DeliveryPointFactory
+from apps.core.models import DeliveryPoint
 
 
 @pytest.mark.django_db
-def test_create_consent():
+def test_create_consent(patch_datetime_now):
     """Tests the creation of a consent."""
-    user1 = UserFactory()
-    delivery_point = DeliveryPointFactory()
+    from apps.consent.factories import ConsentFactory
+    from apps.consent.models import Consent
+
+    assert Consent.objects.count() == 0
+
+    # create one `consent`
+    signals.post_save.disconnect(
+        receiver=handle_new_delivery_point,
+        sender=DeliveryPoint,
+        dispatch_uid="handle_new_delivery_point",
+    )
 
+    delivery_point = DeliveryPointFactory()
     consent = ConsentFactory(
-        delivery_point=delivery_point,
-        created_by=user1,
+        delivery_point=delivery_point, created_by=delivery_point.entity.users.first()
+    )
+
+    signals.post_save.connect(
+        receiver=handle_new_delivery_point,
+        sender=DeliveryPoint,
+        dispatch_uid="handle_new_delivery_point",
     )
 
+    assert Consent.objects.count() == 1
+
     assert consent.delivery_point == delivery_point
-    assert consent.created_by == user1
+    assert consent.created_by == delivery_point.entity.users.first()
     assert consent.status == AWAITING
     assert consent.revoked_at is None
     assert consent.start is not None
     assert consent.end is not None
 
-    # test consent.end is 90 days later than the consent.start
-    end_date = consent.start + timedelta(days=90)
-    consent_start = formats.date_format(end_date, "Y/m/d")
-    consent_end = formats.date_format(consent.end, "Y/m/d")
-    assert consent_start == consent_end
+    # test consent.end is the last day of the year
+    expected_end_date = consent_end_date()
+    assert consent.end == expected_end_date
 
     # test created_at and updated_at have been updated.
     assert consent.created_at is not None
     assert consent.updated_at is not None
 
 
 @pytest.mark.django_db
-def test_update_consent_status():
-    """Tests updating a consent status."""
-    user1 = UserFactory()
-    delivery_point = DeliveryPointFactory()
+def test_create_consent_with_custom_period_date():
+    """Tests the creation of a consent with a custom period date (`start` / `end`)."""
+    from apps.consent.factories import ConsentFactory
+    from apps.consent.models import Consent
 
-    consent = ConsentFactory(
-        delivery_point=delivery_point,
-        created_by=user1,
+    expected_start_date = make_aware(
+        datetime.datetime.strptime("2024-12-20 17:05:02", "%Y-%m-%d %H:%M:%S")
+    )
```

Comment:
I don't understand why you don't create timezone aware datetime instances using `datetime.datetime()` here.